File: CSocket/app/src/main/java/com/westwhale/csocket/MainActivity.java
package com.westwhale.csocket;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {


    private CSocket mCSocket;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Example of a call to a native method
        TextView tv = findViewById(R.id.sample_text);

        mCSocket = CSocket.getInstance(8080);

        tv.setText(mCSocket.toString());
        mCSocket.start();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mCSocket.stop();
        mCSocket.release();
    }
}

File: PthreadTest/app/src/main/java/org/huihui/pthread/MainActivity.kt
package org.huihui.pthread

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn1.setOnClickListener {
            startPthread()
        }
        btn2.setOnClickListener {
            killPthread()
        }
        btn3.setOnClickListener {
            pthreadErrno()
        }
        btn4.setOnClickListener {
            pthreadMutexUnlock()
        }
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    external fun startPthread()

    external fun killPthread()

    external fun pthreadErrno()

    external fun pthreadMutexUnlock()

    companion object {

        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("native-lib")
        }
    }
}
